app/controllers/application_controller/advanced_search.rb

```diff
@@ -45,7 +45,7 @@ def adv_search_build(model)
 
   def adv_search_button_saveid
     if @edit[:new_search_name].nil? || @edit[:new_search_name] == ""
-      add_flash(_("Search Name is required"), :error)
+      add_flash(_("Search Name is required"), :error) if params[:button] == 'saveit'
```

Comment:
@mzazrivec not sure i understand the purpose of this check, this method is called when save/saveit buttons are pressed, don't we need to show flash message in both cases? https://github.com/ManageIQ/manageiq-ui-classic/blob/d6028b8d5535e82250a454f048a63a02f4e3a469/app/controllers/application_controller/advanced_search.rb#L245-L247

Reply:
IIRC, `save` button is that `Save` button, which takes you from expression editor to the filter save form. `saveit` does the actual filter save.

So in the first case, we don't render any flash message. The above code would just cause the flash message to show in app's log. We want to render flash only in the second case, hence the fix above. Although I'm not 100% sure this is the best fix.

Reply:
@mzazrivec makes sense now, thx for explanation :+1: